src/operations/data_operation.rs

```diff
@@ -0,0 +1,8 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use datafusion::prelude::DataFrame;
+
+#[async_trait]
+pub trait DataOperation {
+    async fn apply(&self, df: DataFrame) -> Result<DataFrame>;
+}
```

Comment:
the DataOperation trait provides an abstraction over all transformations of a given format. this way we can combine things like Query, Sort, etc. in any combination that makes sense for us and they're just a series of data operations applied in a certain order